src/java/org/apache/cassandra/transport/Frame.java

```diff
@@ -301,6 +305,10 @@ public void encode(ChannelHandlerContext ctx, Frame frame, List<Object> results)
 
             results.add(header);
             results.add(frame.body);
+
+            int messageSize = header.writerIndex() + frame.body.writerIndex();
```

Comment:
should use `io.netty.buffer.ByteBuf#readableBytes`.  With a allocator there isn't a promise that the readerIndex is always 0, so its possible to get a ByteBuf pointing to a fragment in memory where the readerIndex is something like 102471081; readableBytes is safer